agent/discovery/cloud.go

```diff
@@ -24,23 +26,23 @@ func (d CloudDiscovery) GetId() string {
 	return d.id
 }
 
-func (d CloudDiscovery) Discover() error {
+func (d CloudDiscovery) Discover() (string, error) {
 	cloudData, err := cloud.NewCloudInstance()
 	if err != nil {
-		return err
+		return fmt.Errorf("%v", err).Error(), err
```

Comment:
You are right, we can remove the 1st parameter in the error cases after changing how we use the result in the main discovery loop